src/textarea.rs

```diff
@@ -2003,6 +2003,27 @@ impl<'a> TextArea<'a> {
         self.cursor
     }
 
+    /// Get the current selection start position. 0-base character-wise (row, col) selection start position.
+    /// ```
+    /// use tui_textarea::TextArea;
+    /// use tui_textarea::CursorMove;
+    ///
+    /// let mut textarea = TextArea::from(["abc"]);
+    /// assert_eq!(textarea.selection_start(), None);
+    ///
+    /// textarea.start_selection();
+    /// assert_eq!(textarea.selection_start(), Some((0, 0)));
+    ///
+    /// textarea.move_cursor(CursorMove::Forward);
+    /// assert_eq!(textarea.selection_start(), Some((0, 0)));
+    ///
+    /// textarea.cancel_selection();
+    /// assert_eq!(textarea.selection_start(), None);
+    /// ```
+    pub fn selection_start(&self) -> Option<(usize, usize)> {
+        self.selection_start
+    }
```

Comment:
I feel the following method is better interface because it is a bit confusing when the cursor is before the start position. I think that users want to obtain the range of the selected text rather than a single (row, col) position.

```rust
pub fn selection_range(&self) -> Option<(usize, usize), (usize, usize)> {
    // Return (row, col) pair of selection range
}
```

What do you think?

Reply:
I think that makes sense!

Reply:
actually, @rhysd `selection_range()` conflicts with the private fn which has a different behavior:
https://github.com/rhysd/tui-textarea/blob/93cfabdc6908aad29602823c8e488e540d1580e3/src/textarea.rs#L1433-L1444

i guess we could call the public fn something like `unordered_selection_range()` or `selection_range_raw()`? although those are a bit unwieldy

Reply:
Oops, please just rename the private method to `selection_raange_` (or `selection_range_impl` if clippy gets unhappy. Then I will rename it to more proper name after merging this PR.